# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1457547468 18000
#      Wed Mar 09 13:17:48 2016 -0500
# Node ID fdb9adcc02f9ead5543b9c77636702e54778ab17
# Parent  29e8d4a922b5a3f78a91d2229d55396ccdefdb91
don't substitute SONAME_FLAGS

For now, omit substitution of SONAME_FLAGS.  If this is added again
later, we should ensure that things like $@ are properly substituted.

* subst-config-vals.in.sh, subst-cross-config-vals.in.sh, build-env.h,
corefcn/toplev.cc: Eliminate SONAME_FLAGS.

diff --git a/build-aux/subst-config-vals.in.sh b/build-aux/subst-config-vals.in.sh
--- a/build-aux/subst-config-vals.in.sh
+++ b/build-aux/subst-config-vals.in.sh
@@ -151,17 +151,16 @@ RDYNAMIC_FLAG="@RDYNAMIC_FLAG@"
 READLINE_LIBS="@READLINE_LIBS@"
 SED="@SED@"
 SHARED_LIBS="@SHARED_LIBS@"
 SHLEXT="@SHLEXT@"
 SHLLINKEXT="@SHLLINKEXT@"
 SHLEXT_VER="@SHLEXT_VER@"
 SH_LD="@SH_LD@"
 SH_LDFLAGS="@SH_LDFLAGS@"
-SONAME_FLAGS="@SONAME_FLAGS@"
 STATIC_LIBS="@STATIC_LIBS@"
 TERM_LIBS="@TERM_LIBS@"
 UMFPACK_CPPFLAGS="@UMFPACK_CPPFLAGS@"
 UMFPACK_LDFLAGS="@UMFPACK_LDFLAGS@"
 UMFPACK_LIBS="@UMFPACK_LIBS@"
 version="@PACKAGE_VERSION@"
 WARN_CFLAGS="@WARN_CFLAGS@"
 WARN_CXXFLAGS="@WARN_CXXFLAGS@"
@@ -309,17 +308,16 @@ config_opts="@config_opts@"
   -e "s|%OCTAVE_CONF_READLINE_LIBS%|\"${READLINE_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_SED%|\"${SED}\"|" \
   -e "s|%OCTAVE_CONF_SHARED_LIBS%|\"${SHARED_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_SHLEXT%|\"${SHLEXT}\"|" \
   -e "s|%OCTAVE_CONF_SHLLINKEXT%|\"${SHLLINKEXT}\"|" \
   -e "s|%OCTAVE_CONF_SHLEXT_VER%|\"${SHLEXT_VER}\"|" \
   -e "s|%OCTAVE_CONF_SH_LD%|\"${SH_LD}\"|" \
   -e "s|%OCTAVE_CONF_SH_LDFLAGS%|\"${SH_LDFLAGS}\"|" \
-  -e "s|%OCTAVE_CONF_SONAME_FLAGS%|\"${SONAME_FLAGS}\"|" \
   -e "s|%OCTAVE_CONF_STATIC_LIBS%|\"${STATIC_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_TERM_LIBS%|\"${TERM_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_UMFPACK_CPPFLAGS%|\"${UMFPACK_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_UMFPACK_LDFLAGS%|\"${UMFPACK_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_UMFPACK_LIBS%|\"${UMFPACK_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_VERSION%|\"${version}\"|" \
   -e "s|%OCTAVE_CONF_WARN_CFLAGS%|\"${WARN_CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_WARN_CXXFLAGS%|\"${WARN_CXXFLAGS}\"|" \
diff --git a/build-aux/subst-cross-config-vals.in.sh b/build-aux/subst-cross-config-vals.in.sh
--- a/build-aux/subst-cross-config-vals.in.sh
+++ b/build-aux/subst-cross-config-vals.in.sh
@@ -155,17 +155,16 @@ RDYNAMIC_FLAG="@RDYNAMIC_FLAG@"
 READLINE_LIBS="@READLINE_LIBS@"
 SED="@SED@"
 SHARED_LIBS="@SHARED_LIBS@"
 SHLEXT="@SHLEXT@"
 SHLLINKEXT="@SHLLINKEXT@"
 SHLEXT_VER="@SHLEXT_VER@"
 SH_LD="@SH_LD@"
 SH_LDFLAGS="@SH_LDFLAGS@"
-SONAME_FLAGS="@SONAME_FLAGS@"
 STATIC_LIBS="@STATIC_LIBS@"
 TERM_LIBS="@TERM_LIBS@"
 UMFPACK_CPPFLAGS="@UMFPACK_CPPFLAGS@"
 UMFPACK_LDFLAGS="@UMFPACK_LDFLAGS@"
 UMFPACK_LIBS="@UMFPACK_LIBS@"
 version="@PACKAGE_VERSION@"
 WARN_CFLAGS="@WARN_CFLAGS@"
 WARN_CXXFLAGS="@WARN_CXXFLAGS@"
@@ -313,17 +312,16 @@ config_opts="@config_opts@"
     -e "s|%OCTAVE_CONF_READLINE_LIBS%|\"${READLINE_LIBS}\"|" \
     -e "s|%OCTAVE_CONF_SED%|\"${SED}\"|" \
     -e "s|%OCTAVE_CONF_SHARED_LIBS%|\"${SHARED_LIBS}\"|" \
     -e "s|%OCTAVE_CONF_SHLEXT%|\"${SHLEXT}\"|" \
     -e "s|%OCTAVE_CONF_SHLLINKEXT%|\"${SHLLINKEXT}\"|" \
     -e "s|%OCTAVE_CONF_SHLEXT_VER%|\"${SHLEXT_VER}\"|" \
     -e "s|%OCTAVE_CONF_SH_LD%|\"${SH_LD}\"|" \
     -e "s|%OCTAVE_CONF_SH_LDFLAGS%|\"${SH_LDFLAGS}\"|" \
-    -e "s|%OCTAVE_CONF_SONAME_FLAGS%|\"${SONAME_FLAGS}\"|" \
     -e "s|%OCTAVE_CONF_STATIC_LIBS%|\"${STATIC_LIBS}\"|" \
     -e "s|%OCTAVE_CONF_TERM_LIBS%|\"${TERM_LIBS}\"|" \
     -e "s|%OCTAVE_CONF_UMFPACK_CPPFLAGS%|\"${UMFPACK_CPPFLAGS}\"|" \
     -e "s|%OCTAVE_CONF_UMFPACK_LDFLAGS%|\"${UMFPACK_LDFLAGS}\"|" \
     -e "s|%OCTAVE_CONF_UMFPACK_LIBS%|\"${UMFPACK_LIBS}\"|" \
     -e "s|%OCTAVE_CONF_VERSION%|\"${version}\"|" \
     -e "s|%OCTAVE_CONF_WARN_CFLAGS%|\"${WARN_CFLAGS}\"|" \
     -e "s|%OCTAVE_CONF_WARN_CXXFLAGS%|\"${WARN_CXXFLAGS}\"|" \
diff --git a/libinterp/build-env.h b/libinterp/build-env.h
--- a/libinterp/build-env.h
+++ b/libinterp/build-env.h
@@ -151,17 +151,16 @@ namespace octave
     extern const char *RDYNAMIC_FLAG;
     extern const char *READLINE_LIBS;
     extern const char *SED;
     extern const char *SHARED_LIBS;
     extern const char *SHLEXT;
     extern const char *SHLEXT_VER;
     extern const char *SH_LD;
     extern const char *SH_LDFLAGS;
-    extern const char *SONAME_FLAGS;
     extern const char *STATIC_LIBS;
     extern const char *TERM_LIBS;
     extern const char *UMFPACK_CPPFLAGS;
     extern const char *UMFPACK_LDFLAGS;
     extern const char *UMFPACK_LIBS;
     extern const char *WARN_CFLAGS;
     extern const char *WARN_CXXFLAGS;
     extern const char *X11_INCFLAGS;
diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -1579,17 +1579,16 @@ specified option.\n\
       { "RDYNAMIC_FLAG", octave::build_env::RDYNAMIC_FLAG },
       { "READLINE_LIBS", octave::build_env::READLINE_LIBS },
       { "SED", octave::build_env::SED },
       { "SHARED_LIBS", octave::build_env::SHARED_LIBS },
       { "SHLEXT", octave::build_env::SHLEXT },
       { "SHLEXT_VER", octave::build_env::SHLEXT_VER },
       { "SH_LD", octave::build_env::SH_LD },
       { "SH_LDFLAGS", octave::build_env::SH_LDFLAGS },
-      { "SONAME_FLAGS", octave::build_env::SONAME_FLAGS },
       { "STATIC_LIBS", octave::build_env::STATIC_LIBS },
       { "TERM_LIBS", octave::build_env::TERM_LIBS },
       { "UMFPACK_CPPFLAGS", octave::build_env::UMFPACK_CPPFLAGS },
       { "UMFPACK_LDFLAGS", octave::build_env::UMFPACK_LDFLAGS },
       { "UMFPACK_LIBS", octave::build_env::UMFPACK_LIBS },
       { "WARN_CFLAGS", octave::build_env::WARN_CFLAGS },
       { "WARN_CXXFLAGS", octave::build_env::WARN_CXXFLAGS },
       { "X11_INCFLAGS", octave::build_env::X11_INCFLAGS },
